src/Approximations/overapproximate.jl

```diff
@@ -1736,3 +1736,50 @@ function overapproximate(CHA::ConvexHullArray{N, <:AbstractZonotope{N}}, ::Type{
         return Zaux
     end
 end
+
+"""
+    overapproximate(Z::Zonotope, ::Type{<:HParallelotope}, indices=1:dim(Z))
+
+Overapproximation of a zonotope with a parallelotopic set in constraint
+representation.
+
+### Input
+
+- `Z` -- zonotope
+- `HParallelotope` -- type for dispatch
+- `indices` -- (optional; default: `1:dim(Z)`) generator indices selected when
+                constructing the parallelotope
+
+### Output
+
+An overapproximation of the given zonotope using a parallelotope.
+
+### Algorithm
+
+The algorithm is based on Proposition 8 discussed in Section 5 of [1].
+
+[1] Althoff, M., Stursberg, O., & Buss, M. (2010). *Computing reachable sets of
+hybrid systems using a combination of zonotopes and polytopes*. Nonlinear
+analysis: hybrid systems, 4(2), 233-249.
+"""
+function overapproximate(Z::Zonotope, ::Type{<:HParallelotope}, indices=1:dim(Z))
+    Zred = _overapproximate_hparallelotope(Z, HParallelotope, indices)
+    return convert(HParallelotope, Zred)
+end
+
+function _overapproximate_hparallelotope(Z::Zonotope, ::Type{<:HParallelotope}, indices=1:dim(Z))
+    length(indices) == dim(Z) || throw(ArgumentError("the number of generator indices is $(length(indices)), " *
+                                                     "but it was expected to be $(dim(Z))"))
+
+    p, n = ngens(Z), dim(Z)
+    if p == n
+        return _convert(HParallelotope, Z)
```

Comment:
in fact this should be `Z`

Reply:
But the return type should be `HParallelotope`...?

Reply:
The conversion is done here: https://github.com/JuliaReach/LazySets.jl/pull/1628/files/de8f9b143941a0ffde1f02e9f5c1ccd4a6024e44#diff-eb4fb67940b1678df4462dad2b59a25cR1767